src/main/java/com/voodoodyne/jackson/jsog/JSOGRefDeserializer.java

```diff
@@ -17,7 +17,7 @@
 {
 	@Override
 	public JSOGRef deserialize(JsonParser jp, DeserializationContext ctx) throws IOException, JsonProcessingException {
-		JsonNode node = jp.readValueAsTree();
+		JsonNode node = ctx.readValue(jp, JsonNode.class);
```

Comment:
`ctx.readTree(jp)` method could be also used here, but it was introduced in Jackson 2.10 and the feature (FAIL_ON_TRAILING_TOKES) in 2.9, so let's make Jackson 2.9 users happy is for some reasons they use FAIL_ON_TRAIING_TOKENS and are forced to stay with 2.9.